src/main/java/com/bootcamp/demo/data_access/AccountRepository.java

```diff
@@ -0,0 +1,94 @@
+package com.bootcamp.demo.data_access;
+
+import com.bootcamp.demo.model.Account;
+import com.google.api.core.ApiFuture;
+import com.google.cloud.firestore.*;
+import com.google.firebase.cloud.FirestoreClient;
+import org.springframework.stereotype.Repository;
+
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+@Repository
+public class AccountRepository implements Repo<Account> {
+    @Override
+    public Account findOne(String username) throws ExecutionException, InterruptedException {
+        // return optional?
+        Firestore db = FirestoreClient.getFirestore();
+        CollectionReference collectionReference = db.collection("accounts");
+        DocumentReference documentReference = collectionReference.document(username);
+        ApiFuture<DocumentSnapshot> future = documentReference.get();
+        DocumentSnapshot documentSnapshot = future.get();
+//        if(documentSnapshot.exists())
+//            return documentSnapshot.toObject(Account.class);
+        if (documentSnapshot.exists()) {
+            Map<String, Object> data = documentSnapshot.getData();
```

Comment:
Can't we map the document to an Account without extracting properties one by one?